### ISSUE TITLE
definition of strain

### ISSUE THREAD

**Issue by gabor1** [libAtoms/matscipy#12]
hi Guys!

Vikram and I were reviewing the code in get_delta_plus_epsilon(), because we wanted to understand exactly what was being calculated. Here are some suggestions:

1) you compute the deformation matrix, which is indeed delta+epsilon, as suggested by the name of the function, but then the name of the variable into which it is accumulated is just called epsilon, which is confusing.

2) The deformation matrix that is thus computed is not symmetric, so can include rotations. we suggest to take those out, using M.T*M

3) To actually get a strain, one has to subtract the identity, so we suggest that is done, so the final thing returned should be 0.5 *(M.T*M - I) where M is the thing accumulated in get_delta_plus_epsilon()

**Comment by pastewka** [comment]
I agree that the name Delta+epsilon is confusing because it suggests small strain. It should probably be called deformation gradient. However, I think conversion to Green-Lagrangian strain should be up to the user - but adding a function that returns that would probably be useful.

---
Prof. Lars Pastewka
University of Freiburg
Department of Microsystems Engineering
Office: +49 761 203-67480
http://www.imtek.de/laboratories/simulation
________________________________
Von: gabor1 <notifications@github.com>
Gesendet: Freitag, 20. Juli 2018 16:48
An: libAtoms/matscipy
Cc: Subscribed
Betreff: [libAtoms/matscipy] definition of strain (#12)


hi Guys!

Vikram and I were reviewing the code in get_delta_plus_epsilon(), because we wanted to understand exactly what was being calculated. Here are some suggestions:

  1.  you compute the deformation matrix, which is indeed delta+epsilon, as suggested by the name of the function, but then the name of the variable into which it is accumulated is just called epsilon, which is confusing.

  2.  The deformation matrix that is thus computed is not symmetric, so can include rotations. we suggest to take those out, using M.T*M

  3.  To actually get a strain, one has to subtract the identity, so we suggest that is done, so the final thing returned should be 0.5 (M.TM - I) where M is the thing accumulated in get_delta_plus_epsilon()

-

**Comment by gabor1** [comment]
Sure. Perhaps also rename the function to atomic_deformation_gradient?

-- Gábor
